packages/common/src/config/logger.ts

```diff
@@ -0,0 +1,30 @@
+import { Logger, LoggerOptions, createLogger, format, transports } from 'winston';
+import { LoggerConfig } from './config.js';
+
+export const getLogger = (config: LoggerConfig): Logger => {
+	const transportList: LoggerOptions['transports'] = [];
+
+	const { combine, timestamp, simple, colorize, printf } = format;
+
+	// console transport
+	const consoleLog = new transports.Console({
+		format: combine(
+			timestamp(),
+			colorize(),
+			printf(({ timestamp, level, message }) => `${timestamp} [${level}]: ${message}`),
+		),
+	});
+	transportList.push(consoleLog);
+
+	// file transport
+	if (config.file) {
+		const fileLog = new transports.File({ filename: 'logs.log' });
+		transportList.push(fileLog);
+	}
+
+	const logger = createLogger({
+		level: config.level || 'info',
+		transports: transportList,
```

Comment:
rather than pushing things into an empty array, we could have this line be something that looks more like
```suggestion
		transports: [createConsoleLogger(), createFileLogger()],
```

Reply:
the reason of pushing into an empty array is that there is condition whether or not adding `File transport log`.  We're checking the condition out of the array